src/clj/rems/db/entitlements.clj

```diff
@@ -66,33 +70,59 @@
                          {:status "exception"}))
             status (:status response)]
         (when-not (= 200 status)
-          (log/warnf "Post failed: %s", response))
-        (db/log-entitlement-post! {:target target :payload json-payload :status status})))))
+          (log/warnf "Entitlement post failed: %s", response)
+          (str "failed: " status))))))
 
-(defn- get-entitlements-by-user [application-id]
-  (->> (db/get-entitlements {:application application-id :is-active? true})
-       (group-by :userid)
-       (map (fn [[userid rows]]
-              [userid (set (map :resourceid rows))]))
-       (into {})))
+;; TODO argh adding these everywhere sucks
+(defn- fix-entry-from-db [entry]
+  (update-in entry [:outbox/entitlement-post :action] keyword))
+
+(defn process-outbox! []
+  (doseq [entry (mapv fix-entry-from-db
+                      (outbox/get-entries {:type :entitlement-post :due-now? true}))]
+    ;; TODO could bundle entitlements when sending
+    (if-let [error (post-entitlements! (:outbox/entitlement-post entry))]
+      (let [entry (outbox/attempt-failed! entry error)]
+        (when (not (:outbox/next-attempt entry))
+          (log/warn "all attempts to send entitlement post id " (:outbox/id entry) "failed")))
+      (outbox/attempt-succeeded! (:outbox/id entry)))))
+
+(mount/defstate entitlement-poller
+  :start (scheduler/start! process-outbox! (.toStandardDuration (time/seconds 10)))
+  :stop (scheduler/stop! entitlement-poller))
+
+(defn- add-to-outbox! [action entitlements]
+  (outbox/put! {:outbox/type :entitlement-post
+                :outbox/deadline (time/plus (time/now) (time/days 1)) ;; hardcoded for now
+                :outbox/backoff (.toStandardDuration (time/seconds 10)) ;; ditto
```

Comment:
`:outbox/backoff` is not needed here, because `rems.db.outbox/put!` will anyways overwrite it.

Reply:
good catch